cli/cache.rs

```diff
@@ -3,51 +3,112 @@
 use crate::disk_cache::DiskCache;
 use crate::errors::get_error_class_name;
 use crate::file_fetcher::FileFetcher;
+use crate::http_cache::url_to_filename;
 
+use deno_core::anyhow::anyhow;
 use deno_core::error::AnyError;
 use deno_core::futures::future;
 use deno_core::futures::FutureExt;
 use deno_core::serde::Deserialize;
 use deno_core::serde::Serialize;
 use deno_core::serde_json;
+use deno_core::url::Host;
+use deno_core::url::Url;
 use deno_core::ModuleSpecifier;
 use deno_graph::source::CacheInfo;
 use deno_graph::source::LoadFuture;
 use deno_graph::source::LoadResponse;
 use deno_graph::source::Loader;
 use deno_runtime::permissions::Permissions;
 use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::path::Component;
+use std::path::Path;
+use std::path::PathBuf;
+use std::path::Prefix;
 use std::sync::Arc;
 
-#[derive(Debug, Deserialize, Serialize)]
-pub struct EmitMetadata {
-  pub version_hash: String,
-}
```

Comment:
An alternative solution is to store hashes of all the cached files here as well. This would be more complex though.